api/v1alpha1/openstackserver_types.go

```diff
@@ -104,6 +104,10 @@ type OpenStackServerSpec struct {
 	// be injected into the server instance.
 	// +optional
 	UserDataRef *corev1.LocalObjectReference `json:"userDataRef,omitempty"`
+
+	// SchedulerHintAdditionalProperties are arbitrary scheduler hint key/values.
+	// +optional
+	SchedulerHintAdditionalProperties map[string]string `json:"schedulerHintAdditionalProperties,omitempty"`
```

Comment:
seems this is not true, we can't expect it is always a string from the API spec?

https://docs.openstack.org/api-ref/compute/

```


    The request validation schema is per hint. For example, some require a single string value, and some accept a list of values.

    Hints are only used based on the cloud scheduler configuration, which varies per deployment.

    Hints are pluggable per deployment, meaning that a cloud can have custom hints which may not be available in another cloud.

For these reasons, it is important to consult each cloud’s user documentation to know what is available for scheduler hints.
```

Reply:
@jichenjc yes, it is true.
strictly speaking, `SchedulerHintAdditionalProperties` presents `os:scheduler_hints["additionalProperties"]`, But it is also any type. My concern is i cannot use `map[string]interface{}` because kubebuilder doesn't support it.
I couldn't find any other reference or type struct def which handles `any` type, and i used a `string` here because our custom filter accepts boolean and string type values.

Reply:
@jichenjc do you have thought on this? :)

Reply:
no solution but just questioning the API and the way of implementation
if we are sure the string (maybe define our own format) , or we claim we only support string type filter
it might be ok...

Reply:
i am not sure if any changes required for this convo